core/modules/user/user.theme.inc

```diff
@@ -104,21 +104,42 @@ function theme_user_admin_permissions($variables) {
 
   $roles = user_roles();
   foreach (element_children($form['permission']) as $key) {
-    $row = array();
-    // Module name
+    $row = array(
+      'data' => array(),
+    );
+
+    // Hide the permission (or module) name and description element, since the
+    // individual properties will be pulled out in each row.
+    hide($form['permission'][$key]);
+
+    // Module name row.
     if (is_numeric($key)) {
-      $row[] = array('data' => backdrop_render($form['permission'][$key]), 'class' => array('module'), 'id' => 'module-' . $form['permission'][$key]['#id'], 'colspan' => count($form['roles']['#value']) + 1);
+      $module_text = '<div class="permission-module-name table-filter-text-source">' . $form['permission'][$key]['#markup'] . '</div>';
+      $row['class'] = array('module-row');
+      $row['data'][] = array(
+        'data' => $module_text,
+        'class' => array('module'),
+        'id' => 'module-' . $form['permission'][$key]['#id'],
+        'colspan' => count($form['roles']['#value']) + 1,
+      );
     }
+    // Permission row.
     else {
-      // Permission row.
-      $row[] = array(
-        'data' => backdrop_render($form['permission'][$key]),
-        'class' => array('permission'),
+      $permission_text = '<div class="permission-name table-filter-text-source">' . $form['permission'][$key]['#markup'] . '</div>';
+      $permission_text .= '<div class="permission-description">' . $form['permission'][$key]['#description'] . '</div>';
```

Comment:
This results in unnecessary nested divs, and removes the previous `'description'` class from the text (which gives it the smaller font size and gray color). Can we instead do something like this here (avoid adding any extra divs around the description):

```suggestion
      $permission_text .= $form['permission'][$key]['#description'];
```

...and then add the `'table-filter-text-source'` class either directly in `theme_user_permission_description()`? ...or try passing it as parameter in the `$variables` via `user_admin_permissions()`?

The above assumes that the `'description'` class is added automatically in the resulting div - if not, then that will also need to be added along with `'table-filter-text-source'`.

In general, this is what we had before:
```html
<td class="permission">
  <div id="edit-administer-blocks" class="form-item form-type-item">
    PERMISSION NAME
    <div class="description">PERMISSION DESCRIPTION</div>
  </div>
</td>
```
This is what we get with this PR:

```html
<td class="permission table-filter-text" id="module-edit-upload-editor-images">
  <div class="permission-name table-filter-text-source">PERMISSION NAME</div>
  <div class="permission-description">
    <div class="table-filter-text-source">PERMISSION DESCRIPTION</div>
  </div>
</td>
```

...and this is what we want ideally:
```html
<td class="permission table-filter-text" id="module-edit-upload-editor-images">
  <div class="permission-name table-filter-text-source">PERMISSION NAME</div>
  <div class="permission-description description table-filter-text-source">PERMISSION DESCRIPTION</div>
</td>
```

Reply:
...not sure if we need to retain the `form-item` and `form-type-item` classes in the permission name div like before.

And another diff to consider: the CSS ID was previously being added to the div with the permission name, and it was the name of the permission. Now, it is being added in the `<td>` that contains the divs with the permission name and description, and it is prefixed with `module-`. Not sure what these changes may break.

Reply:
Thanks @klonos, great catches. I updated the PR:

* Description markup updated as you suggest, with the applied styling now staying the same.
* Removed `module-` prefix from permission rows. Now the IDs are the same as before but they are on the rows instead of on a div.